[Merge] 完善学生模块：新增学生信息接口及缓存优化

fanfan187 opened the pull request:
本次 PR 主要完善了学生模块，新增并优化了学生信息的增删改查、禁用操作及相关缓存处理，同时引入班级映射 DTO 以确保数据一致性。

---

## 变更内容

- **新增/修改/删除**：
    - **新增接口**：
        - 在 `StudentController` 中新增了获取单个学生信息（GET /{student_uuid}）、获取学生列表（GET /list）、新增学生（POST）、编辑学生（PUT /{student_uuid}）、禁用学生（PUT /disable/{student_uuid}）以及删除学生（DELETE /{student_uuid}）的接口。
    - **常量更新**：
        - 在 `StringConstant` 中新增了学生不存在和保存失败的错误提示常量，以及用于学生 ID 和姓名校验的正则表达式常量。
    - **数据传输对象**：
        - 新增了 `ClassMappingDTO`、`StudentDisableDTO` 与 `StudentVO`，分别用于班级映射、禁用操作返回及前端传输学生信息。
    - **DAO 层优化**：
        - 在 `StudentDAO` 中增加了 `listStudents` 方法（支持缓存查询）和 `editStudent` 方法（基于事务更新及缓存刷新），以提高数据访问效率。
    - **业务逻辑调整**：
        - `StudentLogic` 实现了 `StudentService` 接口中新增的所有学生操作方法（包括新增、编辑、禁用、删除、查询单条及列表数据），并整合了班级映射信息填充逻辑。
    - **辅助方法**：
        - `UserDAO` 新增 `existsByUserUuid` 方法，用于校验学生对应用户是否已注册。
    - **单元测试**：
        - 在 `StudentTest` 中新增了针对学生列表查询（含缓存验证）和编辑操作的测试用例。

- **涉及模块**：
    - 后端 API 接口、Controller 层、Service 层、DAO 层、缓存模块（Redisson）、数据传输层（DTO/VO）、单元测试

- **关键实现**：
    - 使用 Hutool 的 `BeanUtil` 简化对象属性拷贝；
    - 利用 MyBatis-Plus 的 `LambdaQueryChainWrapper` 进行灵活查询；
    - 通过 Redisson 实现对学生列表查询结果的缓存管理，提升查询性能；
    - 采用事务管理保障编辑操作的一致性，并在操作成功后更新缓存；
    - 在新增和编辑操作中通过调用班级映射接口自动填充年级、部门和专业信息，确保数据完整。

---

## 测试内容

- **单元测试**：
    - 针对 `StudentDAO` 的 `listStudents` 方法测试了缓存命中及数据一致性；
    - 针对 `editStudent` 方法验证了事务操作、缓存更新以及异常回滚功能。

- **集成测试**：
    - 通过接口调用（Postman 或自动化测试脚本）对 `StudentController` 的各个新增接口进行了完整流程验证，包括数据校验、异常处理及返回结果正确性。

- **手动测试**：
    - 在开发环境中对新增、编辑、禁用和删除学生操作进行了手动验证，确保各接口在不同场景下均能正确响应。
    - 特别测试了用户状态与学生禁用状态同步更新的逻辑。

- **新增或修改的测试用例**：
    - 新增了对 `listStudents` 缓存及分页查询效果的测试；
    - 增加了对 `editStudent` 事务处理和缓存刷新场景的验证，测试步骤包括数据修改、缓存验证及异常场景重现。

---

## 相关问题

- **问题 1**：原有学生新增或编辑时未自动填充班级映射信息，可能导致数据不一致；
- **问题 2**：学生禁用操作时未同步更新对应用户的状态，影响后续业务逻辑；

---

fanfan187 committed:
feat(models): 添加学生视图对象

- 新增 StudentVO 类，用于表示学生信息
- 添加了学号、姓名、性别、学院、专业、班级等字段
- 使用了数据校验注解，确保数据的完整性

fanfan187 committed:
feat(StudentController): 添加学生管理功能

- 新增 StudentController 类，实现学生管理相关接口
-包含学生信息查看、学生列表查询、添加学生、停用学生、删除学生和编辑学生等功能

fanfan187 committed:
feat(student): 添加学生信息相关功能

- 新增学生信息查看接口
- 实现学生信息添加功能
- 增加参数校验逻辑
- 定义学生信息相关的正则表达式常量

fanfan187 committed:
Merge remote-tracking branch 'origin/develop' into feature/student-operate

fanfan187 committed:
Merge remote-tracking branch 'origin/develop' into feature/student-operate

fanfan187 committed:
Merge remote-tracking branch 'origin/develop' into feature/student-operate

# Conflicts:
#	src/main/java/com/frontleaves/scheduling/constants/StringConstant.java

fanfan187 committed:
refactor(scheduling): 重构学生相关字段并更新验证注解
- 将性别字段从 Integer 类型改为 Boolean 类型
- 添加 gradeUuid 和 userUuid 字段- 更新字段验证注解，增加不能为空的验证
- 移除多余的 Min 和 Max 注解

fanfan187 committed:
feat(student): 实现学生信息管理功能

- 添加学生信息列表查询接口，支持分页和多条件筛选
- 实现学生信息添加、编辑和停用功能
- 新增学生相关数据传输对象和视图对象
- 优化学生信息的数据库查询和更新逻辑

fanfan187 committed:
feat(student): 实现了删除学生的逻辑

- 注释掉禁用学生的相关代码
- 实现删除学生功能
- 更新学生逻辑层，增加学生存在性检查
- 调整学生数据对象，移除不必要的字段
- 优化学生视图对象转换逻辑

fanfan187 committed:
feat(student): 启用或禁用学生账户功能

- 新增 disableStudent 方法，根据学生 UUID 和禁用标志更新学生账户状态- 在 StudentDTO 中添加 status 字段表示学生状态
- 更新 StudentController 中的 disableStudent 方法，支持启用或禁用学生账户
- 优化 StudentDAO 和 StudentLogic 中的相关方法，支持学生状态更新

fanfan187 committed:
test(student): 补充学生相关测试用例并优化学生模型

- 在 StudentTest 类中添加测试学生列表和编辑学生信息的方法
- 优化 StudentDO、StudentDTO 和 StudentVO 中的字段命名
- 更新 StudentVO 类，使用 Lombok 注解简化代码

fanfan187 committed:
feat(student): 优化学生信息处理逻辑

- 新增 `ClassMappingDTO` 和 `MappingUtil` 用于班级映射
- 重构 `StudentController`、`StudentDAO`、`StudentLogic 和 `StudentService`
- 优化学生信息添加和编辑流程，自动处理年级、学院和专业信息
- 增加 Redis 缓存支持，提高学生信息查询效率
- 更新相关测试用例

xiaolfeng committed:
Merge branch 'develop' into feature/student-operate

# Conflicts:
#	src/main/java/com/frontleaves/scheduling/constants/StringConstant.java
#	src/main/java/com/frontleaves/scheduling/controllers/StudentController.java
#	src/main/java/com/frontleaves/scheduling/daos/StudentDAO.java
#	src/main/java/com/frontleaves/scheduling/logic/StudentLogic.java
#	src/main/java/com/frontleaves/scheduling/services/StudentService.java

xiaolfeng committed:
refactor(student): 重构学生相关代码

- 移除了 StudentDAO 中的冗余方法
- 更新了 StudentLogic 中的学生相关逻辑
- 调整了 StudentController 和 StudentService 中的导入语句
- 删除了未使用的代码和注释

fanfan187 committed:
refactor(scheduling): 重构学生相关功能

- 将 `MappingUtil` 类移除，直接在 `AdministrativeClassDAO` 中实现 getClassMappingByClazz 方法
- 优化 `StudentDAO` 中的 listStudents 方法，使用 `ProjectUtil`工具类实现缓存逻辑
- 重构 `StudentLogic` 类，优化学生相关操作的实现
- 调整 `StudentController` 中的路由和参数处理方式

xiaolfeng commented on src/main/java/com/frontleaves/scheduling/daos/AdministrativeClassDAO.java:
```diff
@@ -162,6 +165,21 @@ public List<AdministrativeClassDO> getAdministrativeClassListByDepartmentForUpda
+                .setDepartmentUuid(adminClass.getDepartmentUuid())
+                .setMajorUuid(adminClass.getMajorUuid());
+    }
+}
```
1. 文档注释
2. 不建议 DAO 转义后返回数据，数据格式转换推荐在 Logic 进行（方便扩展可以直接复用该接口）
3. 没有使用缓存

xiaolfeng commented on src/main/java/com/frontleaves/scheduling/logic/StudentLogic.java:
```diff
@@ -36,19 +51,193 @@
+     */
+    @Override
+    public StudentDTO editStudent(String studentUuid, StudentVO studentVO) {
+        StudentDO studentDO = studentDAO.editStudent(studentUuid, studentVO);
```
没有验证班级是否存在

fanfan187 committed:
feat(scheduling): 实现行政班级信息的 Redis 缓存

- 在 AdministrativeClassDAO 中添加 getAdministrativeClassMappingByClazz 方法，实现缓存逻辑
- 新增 AdministrativeClassLogic 和 AdministrativeClassService 接口，用于获取班级映射信息
- 修改 StudentDAO 和 StudentLogic 中使用新的缓存方法获取班级信息
- 更新相关测试用例，验证缓存功能的正确性

fanfan187 committed:
Merge branch 'develop' into feature/student-operate

# Conflicts:
#	src/main/java/com/frontleaves/scheduling/daos/UserDAO.java
#	src/test/java/com/frontleaves/scheduling/dao/StudentTest.java

fanfan187 committed:
test: 删除学生测试类中的冗余代码

- 移除了 StudentTest 类中不必要的右大括号
